server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -1035,7 +1035,7 @@ private Pair<List<UserVmJoinVO>, Integer> searchForUserVMsInternal(ListVMsCmd cm
         Object backupOfferingId = cmd.getBackupOfferingId();
         Object isHaEnabled = cmd.getHaEnabled();
         Object pod = null;
-        Long clusterId = null;
+        Object clusterId = null;
         Object hostId = null;
         Object storageId = null;
         if (_accountMgr.isRootAdmin(caller.getId())) {
```

Comment:
@DaanHoogland I am afraid the below code may not work as cmd will be an instance of `ListVMsMetricsCmdByAdmin` and not `ListVMsCmdByAdmin`